### ISSUE TITLE
Significant performance decrease on 5.2.1

### ISSUE THREAD

**Issue by daniluk4000** [nuxt-modules/stylelint#144]
Sorry, but I can't provide reproduction yet, if it is very much needed I will.

First of all, we have noticed many many duplicating errors. Plugin was infinite printing one single error to console.

When we fixed this, I have noticed infinite plugin compiling:

https://github.com/nuxt-modules/stylelint/pull/127 - this has probably caused those issues

**Comment by AdrianFahrbach** [comment]
We got the same issue. It looks like the Vite server is restarting multiple times.
We also got caching issues with styles in that project. So changes weren't detected with hot module reloading (not quite sure though if this is related).
Downgrading to 5.2.0 fixed this for us as well.

**Comment by janvorisek** [comment]
I'm also confirming the issue. Downgrading to 5.2.0 works.

**Comment by gerbenvandijk** [comment]
same issue here. downgrading works but then i get issues with peer dependancies:

```
 WARN  Issues with peer dependencies found
.
├─┬ @nuxtjs/stylelint-module 5.2.0
│ └─┬ vite-plugin-stylelint 5.3.1
│   └── ✕ unmet peer vite@"^2.0.0 || ^3.0.0 || ^4.0.0 || ^5.0.0": found 6.3.3 in @nuxt/vite-builder
```

**Comment by ThomasBerneHCSE** [comment]
I uninstalled the module. It caused really bad hmr and dev server loading time.

With the module:
![Image](https://github.com/user-attachments/assets/640a6f79-bfe7-4599-9b23-e1bf09201320)

Without the module:
![Image](https://github.com/user-attachments/assets/15ec21ea-4f61-4f40-a18b-a8f1396fea23)

**Comment by AdrianFahrbach** [comment]
We did check this again, and it seems like the module is indeed causing major HMR issues, even on v5.2.0. With this module enabled, we always get messages like `(repeated 17 times)` from Vite in the console. Without the module everything works fine.

I guess that this is an issue with the recent upgrade to Vite 6, so this may not affect you on older Nuxt version. With Vite 6 I highly recommend to not use this module though.
